src/core/src/op/avg_pool.cpp

```diff
@@ -19,31 +19,12 @@ AvgPool::AvgPool(const Output<Node>& arg,
                  const Shape& pads_end,
                  const Shape& kernel,
                  bool exclude_pad,
-                 op::RoundingType rounding_type,
-                 const PadType& auto_pad)
-    : Op({arg}),
-      m_kernel(kernel),
-      m_strides(strides),
-      m_pads_begin(pads_begin),
-      m_pads_end(pads_end),
-      m_exclude_pad(exclude_pad),
-      m_auto_pad(auto_pad),
-      m_rounding_type(rounding_type) {
+                 RoundingType rounding_type,
+                 PadType auto_pad)
```

Comment:
```suggestion
                 const PadType& auto_pad)
```
Don't change API, please.

Reply:
Pass by value is fine in for this type, but we cannot just change functions signatures if it is public API. so for AvgPool ops it should be unchanged.
For base class it can be different as this is new class, but maybe is better to align to have same type as current AvgPools. The reference will passed to base class and then copied.

Reply:
Thanks for the explanation!
I fixed it in https://github.com/openvinotoolkit/openvino/pull/23483/commits/4080a715a38a51bde110895f0c1a0055a394221d
Please, review again